### PR TITLE
ci: setup codecov and add coverage badge

### PR DESCRIPTION
**Pull request by sudiptog81**
Fixes #25. Added codecov coverage as an GitHub action and add a badge in readme.md.
Need to update the `CODECOV_TOKEN` in the repo secrets after enabling it in codecov.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ![image](https://user-images.githubusercontent.com/7620947/119247309-b69d1580-bb5e-11eb-9d81-4495dfc45f21.png)
 
 [![tests](https://github.com/rodrigo-brito/ninjabot/actions/workflows/ci.yaml/badge.svg)](https://github.com/rodrigo-brito/ninjabot/actions/workflows/ci.yaml)
+[![codecov](https://codecov.io/gh/sudiptog81/ninjabot/branch/main/graph/badge.svg?token=VfrjGMzXC5)](https://codecov.io/gh/sudiptog81/ninjabot)
```
Comment (rodrigo-brito) [inline readme.md:4]:
```suggestion
[![codecov](https://codecov.io/gh/rodrigo-brito/ninjabot/branch/main/graph/badge.svg)](https://codecov.io/gh/sudiptog81/ninjabot)
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -19,3 +19,8 @@ jobs:
           version: latest
           skip-build-cache: true
           skip-pkg-cache: true
+
+      - name: coverage
+        uses: codecov/codecov-action@v2
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```
Comment (rodrigo-brito) [inline .github/workflows/ci.yaml:26]: how I can include this secret in the project?

**Review by rodrigo-brito — COMMENTED** [review]
Hi @sudiptog81, thank you for the contribution

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (sudiptog81) [reply .github/workflows/ci.yaml:26]: Please go to https://github.com/rodrigo-brito/ninjabot/settings first. The Secrets tab on the left, and add a new repository secret. Use `CODECOV_TOKEN` for the name and the value for the token can be found at https://app.codecov.io/gh/rodrigo-brito/ninjabot/settings/general once you enable it on the codecov frontend.

File: readme.md (hunk 1 above)
Comment (sudiptog81) [reply readme.md:4]: Ah, good catch, was just testing it out. Fixing in next commit.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (rodrigo-brito) [reply .github/workflows/ci.yaml:26]: Ok, done

**Comment by sudiptog81** [comment]
The uploader probably errored as `go test` did not generate the coverage files. Can use `-coverprofile=coverage.txt -covermode=atomic` flags.

**Review by rodrigo-brito — APPROVED** [review]

**Pull request merged**